### PR TITLE
Align destroy capabilities

### LINKED ISSUE #738: Destroy stream button not present for org owners
Relates to #634 

The spreadsheet + our current ability model indicates that org owners can destroy non-default streams of their own org (Owners have the `:crud` ability on Stream). While the intended Destroy button is currently showing up for POD Admins, the button isn't showing up for org owners.

Current button code:
https://github.com/pod4lib/aggregator/blob/f42b15f90c770258cf8b9963d55de8c32ec7955d/app/views/streams/index.html.erb#L31

If you change `if can? :delete, stream` to `if can? :destroy, stream`, the button properly shows up for owners.
I don't know the underlying implications of `:delete` vs `:destroy` 

This ticket should also include looking through the UI to other places where the destroy/delete button is supposed to be showing, and probably writing a test.

### PR DESCRIPTION
**Pull request by marlo-longley**
Fixes #738 

Also found a decent way to test the providers scope on organizations.

### CODE REVIEW

**Comment by marlo-longley** [comment]
Latest commit also fixes #719

File: spec/models/organization_spec.rb [hunk 1]
```diff
@@ -48,4 +48,16 @@
...
+    it 'includes providers in scope' do
+      org = described_class.create!(provider: true)
+      expect(described_class.providers).to include(org)
+    end
+
+    it 'excludes non-providers in scope' do
+      org = described_class.create!(provider: false)
+      expect(described_class.providers).not_to include(org)
+    end
+  end
```
Comment (thatbudakguy) [inline spec/models/organization_spec.rb:62]: this is sort of what i was alluding to in standup — if you define a super simple scope using ActiveRecord on one field, there isn't really a way for it to fail unless Rails's own code fails, so you're effectively testing Rails. not a bad thing necessarily, just generally less useful than other kinds of tests

**Review by thatbudakguy — APPROVED** [review]
pull request title sounds like a metal band. code looks good 👍

**Review by corylown — APPROVED** [review]

File: spec/models/organization_spec.rb (hunk 1 above)
Comment (marlo-longley) [reply spec/models/organization_spec.rb:62]: @thatbudakguy fair! I couldn't think of another way to test it.

Comment (marlo-longley) [reply spec/models/organization_spec.rb:62]: also - destroy permissions!!!

**Pull request merged**